src/value.zig

```diff
@@ -1828,27 +1828,55 @@ pub const Value = extern union {
         return order(lhs, rhs).compare(op);
     }
 
-    /// Asserts the value is comparable. Both operands have type `ty`.
+    /// Asserts the values are comparable. Both operands have type `ty`.
+    /// Vector results will be reduced with AND.
```

Comment:
This function doesn't create a new `Value` and only returns a `bool`, so we can't return a vector in its place. A lot of callsites use `compare` just as a simple test for "if this value is some known constant (i.e. an identity like 0 or 1), then I can do less work", but creating a vector & immediately reducing it with AND would be rather pointless and require needless allocations. The only time we would ever want to _not_ reduce vector comparisons with AND is when we're handling a Zig comparison instruction. Otherwise, we'll almost always just want a singular scalar to branch off of.